drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -101,28 +102,39 @@ void ng_at86rf2xx_reset(ng_at86rf2xx_t *dev)
     /* set short and long address */
 #if CPUID_ID_LEN
     cpuid_get(cpuid);
-#if CPUID < 8
+#if CPUID_ID_LEN < 8
     /* in case CPUID_ID_LEN < 8, fill missing bytes with zeros */
     for (int i = CPUID_ID_LEN; i < 8; i++) {
         cpuid[i] = 0;
     }
+    /* copy and set long address */
+    memcpy(&addr_long, cpuid, 8);
 #else
-    for (int i = 8; i < CPUID_ID_LEN; i++) {
-        cpuid[i & 0x07] ^= cpuid[i];
-    }
+    /* for consistency taken from OpenWSN, https://github.com/openwsn-berkeley/openwsn-fw,
+     * bsp/boards/iot-lab_M3/eui64.c */
+    const uint8_t const *luid = (const uint8_t *const) &(cpuid[CPUID_ID_LEN - 8]);
+
+    /* Pseudo-OUI */
+    addr_long.uint8[0] = cpuid[3];
+    addr_long.uint8[1] = luid[7];
+    addr_long.uint8[2] = cpuid[0];
+    addr_long.uint8[3] = cpuid[1];
```

Comment:
Because Atmel is handling its device identifiers => a collision is bound to happen. And I don't know how the legal situation is. Also: since it is still a pseudo EUI-64 this way the correct prefix would be 020425, since the universal/local-bit must be set.